containers/unit_tests/TestBitset.hpp

```diff
@@ -156,7 +156,7 @@ void test_bitset() {
   {
     unsigned ts = 100u;
     bitset_type b1;
-    ASSERT_TRUE(b1.is_allocated());
+    ASSERT_FALSE(b1.is_allocated());
```

Comment:
So Bitset is always allocated now even if the size is zero?

Reply:
I think it's the contrary. Before this PR, `Bitset` default construction would allocate its member with a zero-size (`Bitset(unsigned arg_size = 0u)`). Now, the default constructor does nothing, so by default it is not allocated. However, I think the behavior when constructing a `Bitset` of size `0` is unchanged.

Reply:
The changes related to `Bitset` allocation have been reverted, and a discussion opened in #6467.